### DISCUSSION TITLE
Description/tooltip when hovering over items?

### DISCUSSION THREAD

**Discussion by kyle-wannacott** [peter-kish/gloot#185] [Q&A] [answered]
How would you recommend doing a description box when hovering over an item?

Control nodes have a tooltip, so could use that for items in ItemSlots, however it seems very limited (like I don't think you can color text differently in it). I think I would prefer something a bit more customizable.

Is there a way to get whether you are hovering over an item in the InventoryGrid? (and ItemSlot?) (I can't see anything related to this in the docs)?

I guess if  there was a signal that emitted when someone was hovering over an item (which told you what item it is); then I could build a custom description box for it and place it near the cursor. (probably also needs signal when exit the hover)

Any ideas, or comments?

Like for example description when hovering over an item in Diablo 2:
![image](https://github.com/peter-kish/gloot/assets/49783296/e689ddb6-3965-4826-8bcc-72b3513b9ae0)

**Comment by peter-kish** [comment, marked as answer]
You mean something like the `item_mouse_entered` and `item_mouse_exited` signals in [CtrlInventoryGrid](https://github.com/peter-kish/gloot/blob/master/docs/ctrl_inventory_grid.md)? :)

**Reply by kyle-wannacott** [reply]
Ahh yeah, Guess I didn't look hard enough :clown_face:  (I think I was looking at inventory grid rather than the control). Should be do-able then; might need to put an await so the description doesn't pop up instantly. Cheers!

**Reply by Rickyfs** [reply]
Hello, I try inventory_grid_transfer.tscn example which I see in attached script it connects _on_item_mouse_entered and should show labels on item hover however it does not work at all for me. I tried to connect signals manually too but it seems there is no such signal functionality?

**Reply by kyle-wannacott** [reply]
It was working for me in a previous release. Although I think it may have broken?, as I was trying to get it working last night and no signals were being emitted. I thought it was because I refactored my code into a separate file and broke it somehow. @Rickyfs  However; you making this comment makes me think it might be a regression in Gloot ? @peter-kish

**Reply by peter-kish** [reply]
Must be some sort of regression 😬 I'll take a look...

**Reply by peter-kish** [reply]
It was a bug introduced during some refactoring. Should be fixed now in 2db328e35965960ff6d1c8db02ef321cb970e5eb.

**Reply by kyle-wannacott** [reply]
Nice one!, It might be prudent to check the other signals on CtrlInventoryGrid as well; I don't think they are working either...

**Reply by peter-kish** [reply]
Right, I completely missed those. 332d4429071ba6a8c031770bff8d81338fca3e0a should cover it.
Issues like this make me think about writing tests for the UI classes too.
